### ISSUE TITLE
Allow free-form reactions

### ISSUE THREAD

**Issue by tadzik** [vector-im/element-web#19409]
#### What would you like to do?

Post reactions other than emojis.

#### Why would you like to do it?

To unleash my creativity! Also, I hate emojis.

#### How would you like to achieve it?

By merging https://github.com/matrix-org/matrix-react-sdk/pull/6628.

### Have you considered any alternatives?

The only alternative currently available is using another client. Freeform reactions are already used in the wild though, for example by TWIM bot.

**Comment by bkil** [comment]
It would also be neat to have a UX for producing custom characters using Unicode combiners and whatnot.

**Comment by NicolasDerumigny** [comment]
Note that discord bots also sends custom reactions with server-defined emojis; this would be a great feature to display in element.

**Comment by HarHarLinks** [comment]
> server-defined emojis; this would be a great feature to display in element.

as implemented in fluffychat for a while, i think. however doing this proper would probably require a spec change to allow reaction events not just with a text key but a flag to explicitly state that it's a mxc:// media url.

**Comment by bkil** [comment]
Many clients opt to hide images and a great number of users can not see images at all due to visual impairment.

Hence it is considered good practice to not only associate an emoji with a picture, but perhaps to also keep providing alternate text (or a Unicode based approximation) as well.

Server-defined emojis may also raise copyright concerns.

**Comment by HarHarLinks** [comment]
> Many clients opt to hide images and a great number of users can not see images at all due to visual impairment.
> 
> Hence it is considered good practice to not only associate an emoji with a picture, but perhaps to also keep providing alternate text (or a Unicode based approximation) as well.

good point, this sounds like an additional argument to updating the spec for custom reaction emoji support, such that an alt-text is associated with emoji.

> Server-defined emojis may also raise copyright concerns.

more than any image shared by anyone already does?

**Comment by bkil** [comment]
Typing a Unicode emoji can never raise a copyright problem 🆓 ©️ 🐧

If as a user, you are uploading a media file that you will share with others in a given room (copy & publish) that implies that you mush have already obtained a license to allow for that (or the work is copyleft, or you are the original author)

The wording in the above comment seems to imply that a given home server (or would it be matrix.org?) should be hosting some kind of media that you could instantly reproduce with a click when you are chatting. This would imply that they had previously obtained a license for the media that allows unlimited redistribution.

Because Discord is a centralized, closed, for-profit platform, they could use various arrangements that could not even work between open federated servers.

**Comment by HarHarLinks** [comment]
> that implies that you mush have already obtained a license to allow for that (or the work is copyleft, or you are the original author)

same applies when I upload a custom sticker pack or custom emoji pack, no?

> you could instantly reproduce with a click

you already can: click "forward message" on an image

> This would imply that they had previously obtained a license for the media that allows unlimited redistribution.

wouldn't I as the sticker/emoji pack author be liable for that?

this discussion is getting a bit off-topic and is probably more suited for the actual MSC: https://github.com/matrix-org/matrix-doc/pull/1951

**Comment by bkil** [comment]
I'm more interested in free-form reactions than in custom sticker packs at the moment, so I'd like to leave that to you.

**Comment by JokerGermany** [comment]
gomuks is already able to make text reactions
![grafik](https://user-images.githubusercontent.com/30293477/155333458-b513a4e3-54c2-42a3-924c-e1c0f0711efb.png)

**Comment by tadzik** [comment]
@kittykat I resolved the merge conflicts in the PR – is there anything else for me to do here?

**Comment by kittykat** [comment]
@tadzik Thanks for rebasing - we can now test it out in Netlify :+1: 

The next step is to get a design review which takes into account [the considerations Jim and I listed here](https://github.com/matrix-org/matrix-react-sdk/pull/6628#pullrequestreview-968940204) before we can proceed.

Quick look at the Netlify build, I'd expect feedback from design at least about the font in the timeline (note that it's different on the "1" right and left):
![Screenshot from 2022-05-20 10-00-13](https://user-images.githubusercontent.com/51663/169494026-70e4a9b4-460e-4687-8b7c-4c08c549dee5.png)

And the layout of the clickable text in the emoji picker for reactions:
![Screenshot from 2022-05-20 10-00-40](https://user-images.githubusercontent.com/51663/169494079-023cffaa-b131-4bd1-959e-d59866303650.png)

And there's no reporting/moderation mechanism right now. Maybe the report dialog could have a tickbox asking "do you want to include the reactions in your report?". This is just a thought, final say on how this should look will be with design and Jim can help with how it should work.

**Comment by t3chguy** [comment]
> And there's no reporting/moderation mechanism right now. Maybe the report dialog could have a tickbox asking "do you want to include the reactions in your report?". This is just a thought, final say on how this should look will be with design and Jim can help with how it should work.

This wouldn't scale well, the report API takes a single event ID so if an event has 40 reactions then now we need to hit the API 41 times
